### DISCUSSION TITLE
Using Magick.NET on Azure

### DISCUSSION THREAD

**Discussion by isonil** [dlemstra/Magick.NET#911] [Help] [answered]
Hi,

does anyone know if it's possible to use Magic.NET in an ASP.NET application deployed to Azure? I have added these 2 NuGet packages:
- Magick.NET.Core
- Magick.NET-Q8-AnyCPU

This seems to work fine locally. However, it fails while deploying to Azure App Service. It takes a very long time to deploy and then fails with the following message:

```
Command 'starter.cmd "C:\home\site\d ...' was aborted due to no output nor CPU activity for 300 seconds. You can increase the SCM_COMMAND_IDLE_TIMEOUT app setting (or WEBJOBS_IDLE_TIMEOUT if this is a WebJob) if needed.\r\nstarter.cmd "C:\home\site\deployments\tools\deploy.cmd"
_CopyFilesMarkedCopyLocal:
One or more errors occurred.
Command 'starter.cmd "C:\home\site\d ...' was aborted due to no output nor CPU activity for 300 seconds. You can increase the SCM_COMMAND_IDLE_TIMEOUT app setting (or WEBJOBS_IDLE_TIMEOUT if this is a WebJob) if needed.\r\nstarter.cmd "C:\home\site\deployments\tools\deploy.cmd"
```

Thanks in advance,
Piotr

**Comment by dlemstra** [comment, marked as answer]
You can probably get more details in the event viewer of that app service. Have you tried searching the logging?

**Reply by isonil** [reply]
Yes, unfortunately it only said something like "The action was interrupted by user" on timeout. I think maybe it was because of the free Azure account and thus exceeding some limits. Though other nuget packages work correctly.

**Reply by dlemstra** [reply]
Have you tried to use `Magick.NET-Q8-x64` instead? It might be possible that the `AnyCPU` version has problems writing the Native library to a temporary folder.

**Reply by isonil** [reply]
Yes, I have. In fact, this was the first one I tried. Then I tried the "any cpu" version.

**Reply by dlemstra** [reply]
Is it possible to create something small that can be used to reproduce the issue?

**Reply by isonil** [reply]
I believe an empty ASP.NET project with Magick.NET-Q8-x64 nuget package, uploaded to a free Azure WebService account would reproduce it. I suspect maybe it's an issue with the free account - maybe the package is simply too big.

**Reply by dlemstra** [reply]
I really wonder if it is the limits of the free version. I suspect that due to some unknown issue the app service fails to start. Could also be that the native library is missing from the bin folder. If you are able to setup a small test project that reproduces the issue I might be able to help you.

**Reply by isonil** [reply]
Here's a default new project (with Magick.NET) which reproduces the issue.
[TestP.zip](https://github.com/dlemstra/Magick.NET/files/6566254/TestP.zip)

**Reply by dlemstra** [reply]
Thanks will try to take a look at it once my MSDN license issues have been resolved. Not sure when that will happen though.

**Reply by isonil** [reply]
Great, thanks.

**Reply by dlemstra** [reply]
I was able to publish your project to Azure without any issues with the free service plan "(F1: Free)". Maybe you got firewall issues where VisualStudio is not allowed to push the files to Azure?

**Reply by isonil** [reply]
Huh, interesting, thanks for checking! If it actually worked and you could see the default home page then perhaps it's something on my end then. I'm using continuous deployment with Bitbucket. Maybe my app service is just too bloated and there's not enough space or something.

**Reply by dlemstra** [reply]
I saw the default home page. Not sure what is happening on your end.
